bluemira/radiation_transport/radiation_profile.py

```diff
@@ -0,0 +1,454 @@
+
+        n_grad_sepped = self.params["neped"] - self.params["nesep"]
+        t_grad_sepped = self.params["teped"] - self.params["tesep"]
+
+        rho_ratio = (1 - rho_core[i_exterior]) / (1 - self.rho_ped)
+
+        ne_e = self.params["nesep"] + n_grad_sepped * rho_ratio
+        te_e = self.params["tesep"] + t_grad_sepped * rho_ratio
+
+        ne_core = np.append(ne_i, ne_e)
+        te_core = np.append(te_i, te_e)
+
+        return ne_core, te_core
+
+    def calculate_sol_decay(
+        self, t_sep, lambda_q_n=0.003, lambda_q_f=0.1, midplane=True
+    ):
+        """
+        Esponential decay to apply from a generic starting point at the separatrix.
+
+        Parameters
+        ----------
+        t_sep: float
+            initial temperature value at the separatrix
+        lambda_q_n: float
+            near SoL decay lenght. Unit [m]
+        lambda_q_f: float
+            far SoL decay lenght. Unit [m]
+
+        Returns
+        -------
+
+        te_sol: np.array
+        """
+        # flux expansion equal 1 at the midplane.
+        # TODO: calculation of flux expansion for other locations
+        f_exp = 1
+        if midplane is False:
+            # f_exp = 10 temporary value
+            f_exp = 10
+            lambda_q_n = lambda_q_n * f_exp
+            lambda_q_f = lambda_q_f * f_exp
+
+        dr = self.dr_flux_tubes_mp * f_exp
+        # Relation between power decay and temperature decay for conduction-limited regime.
+        lambda_t_near = 7 / 2 * lambda_q_n
+        lambda_t_far = 7 / 2 * lambda_q_f
+        i_sol_near = np.where(dr < lambda_q_n)
+        i_sol_far = np.where(dr > lambda_q_n)
+        te_sol_near = t_sep * np.exp(-dr[i_sol_near] / lambda_t_near)
+        te_sol_far = te_sol_near[-1] * np.exp(-dr[i_sol_far] / lambda_t_far)
+        te_sol = np.append(te_sol_near, te_sol_far)
+
+        return te_sol
+
+    def calculate_sol_electron_temperature_mp(self):
+        """
+        Calculation of electron temperature across the SoL at midplane.
+        Using exponential decay.
+
+        Returns
+        -------
+        te_sol: np.array
+        """
+        te_sep = self.params["tesep"]
+        te_sol = self.calculate_sol_decay(te_sep)
+        return te_sol
+
+    def calculate_div_temperatures(
+        self, firstwall, p_sol=200e6, lambda_q_near=0.003, k_0=2000
+    ):
+        """
+        Calculation of electron temperature across the SoL at the divertor entrance.
+        Using the equation to calculate T(s||).
+
+        Parameters
+        ----------
+        firstwall: loop
+            first wall geometry
+        p_sol: float
+            power entering the SoL. P_sol = P_tot - P_rad_main. Unit [W]
+        lambda_q: float
+            near SoL decay lenght. Unit [m]
+
+        Returns
+        -------
+        t_div: np.array
+        """
+        # t_sep from process is in KeV. I need eV.
+        t_sep_mp = self.params["tesep"] * 1000
```

Comment:
Make sure it is being tested as @je-cook is working on the units

Reply:
At the moment my plan is to have user facing units and between codes units consistent. In the first run through at least I will not be dealing with consistency across the codebase.

If params is a ParameterFrame its units should end up being consistent. I can't guarantee anything else